virtual-desktop/src/app/application-manager/application-manager.service.ts

```diff
@@ -400,6 +400,11 @@ export class ApplicationManager implements MVDHosting.ApplicationManagerInterfac
   killApplication(plugin:ZLUX.Plugin, appId:MVDHosting.InstanceId):void {
     ZoweZLUX.dispatcher.deregisterPluginInstance(plugin,
                                                 appId);   // instanceId is proxy handle to isntance 
+    let filePath : any = 'pluginData' + '/' + 'app'
+    let fileNameToDelete : string = plugin.getIdentifier() + '-' + appId
+    this.http.delete(ZoweZLUX.uriBroker.pluginConfigUri(ZoweZLUX.pluginManager.getDesktopPlugin(),filePath,fileNameToDelete)).subscribe(()=>
```

Comment:
From my comment about using a single file (option 2), then in the case that you close 1 instance, the simplest way to update the autosave data is to send the autosave event to all remaining instances of that plugin, so that you can save the result as a smaller array.

an alternative is to get the current file and remove the index for the instance you just closed. in that way, you would need to have a way to track which index this was, like with a map or some mathematical relationship between the id and the index.

Reply:
Using Option 1